scb-engine/src/main/java/io/securecodebox/engine/service/ProcessService.java

```diff
@@ -0,0 +1,84 @@
+ *
+ *  	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ * /
+ */
+package io.securecodebox.engine.service;
+
+import io.securecodebox.constants.DefaultFields;
+import io.securecodebox.model.execution.Target;
+import io.securecodebox.model.rest.SecurityTest;
+import io.securecodebox.scanprocess.ProcessVariableHelper;
+import org.camunda.bpm.engine.ProcessEngine;
+import org.camunda.bpm.engine.repository.ProcessDefinition;
+import org.camunda.bpm.engine.runtime.ProcessInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component
+public class ProcessService {
+    public static class NonExistentProcessException extends Exception{}
+    public static class DuplicateProcessDefinitionForKeyException extends Exception{}
+
+    @Autowired
+    ProcessEngine engine;
+
+    public void checkProcessExistence(String processKey) throws NonExistentProcessException, DuplicateProcessDefinitionForKeyException{
+        long processCount = engine.getRepositoryService()
+                .createProcessDefinitionQuery()
+                .active()
+                .processDefinitionKey(processKey)
+                .latestVersion()
+                .count();
+
+        if (processCount == 0) {
+            throw new NonExistentProcessException();
+        } else if (processCount > 1) {
+            throw new DuplicateProcessDefinitionForKeyException();
+        }
+    }
+
+    public UUID startProcess(SecurityTest securityTest){
+        Map<String, Object> values = new HashMap<>();
+
+        List<Target> targets = new LinkedList<>();
+        targets.add(securityTest.getTarget());
+
+        values.put(DefaultFields.PROCESS_AUTOMATED.name(), true);
+        values.put(DefaultFields.PROCESS_CONTEXT.name(), securityTest.getContext());
+        values.put(DefaultFields.PROCESS_TARGETS.name(), ProcessVariableHelper.generateObjectValue(targets));
+
+        ProcessInstance instance = engine.getRuntimeService().startProcessInstanceByKey(securityTest.getProcessDefinitionKey(), values);
+        return UUID.fromString(instance.getProcessInstanceId());
+    }
+
+    public List<String> getAvailableProcessKeys(){
+        List<ProcessDefinition> allProcesses = engine.getRepositoryService()
+                .createProcessDefinitionQuery()
+                .active()
+                .latestVersion()
+                .list();
+
+        return allProcesses
+                .stream()
+                .map(ProcessDefinition::getKey)
+                .map(def -> def.replace(SecurityTest.PROCESS_NAME_SUFFIX, ""))
```

Comment:
The `-process` suffix is removed as the name of a "nmap security-test" should not be "nmap-process". This suffix directly points at a implementation detail which isn't relevant for this api. 

I'll refactor this so that this mapping between the security test name and the process is completely contained inside the security-test class.